### PR TITLE
Use destructuring for readability and optimzations at 'for' loop length property…

### PR DESCRIPTION
**Pull request by prashant-andani**
Add destructuring for readability and optimzations at 'for' loop length property…

### CODE REVIEW

File: src/language-yaml/utils.js [hunk 1]
```diff
@@ -81,8 +82,9 @@ function isLastDescendantNode(path) {
     case "nonSpecificTag":
       return false;
   }
+  const pathStackLength = path.stack.length - 1;
```
Comment (ikatyang) [inline src/language-yaml/utils.js]: This one looks not equivalent to the original one.

File: src/language-vue/parser-vue.js [hunk 2]
```diff
@@ -146,9 +147,8 @@ function parseHTML(html, options) {
         }
       }
 
-      let text;
-      let rest;
-      let next;
+      let text, rest, next;
```
Comment (ikatyang) [inline src/language-vue/parser-vue.js]:
```
✘  one-var            Split 'let' declarations into multiple statements                                                                                               

src/language-vue/parser-vue.js:150:7
      let text, rest, next;
       ^
```

**Comment by prashant-andani** [comment]
@ikatyang Thanks for catching it... I have fixed it.

**Comment by ikatyang** [comment]
Thanks!

**Comment by duailibe** [comment]
Did you measure the perf improvements of memoizing `list.length`? I don't think this is actually an improvement in perf but it's less readable IMO

**Comment by ikatyang** [comment]
Agreed that's less readable, the balance between readability and performance is hard, not sure how to deal with it.

https://jsperf.com/for-vs-foreach

image

**Comment by duailibe** [comment]
Funny I get different results.

image

Nevertheless, we should measure this. I've been thinking about adding benchmark tests to measure perf improvements/regressions but haven't had the time lately.

**Pull request merged**